### PR TITLE
Shutdown the node program for sys interrupt too

### PR DESCRIPTION
**Pull request by rlan35**
Our validators are used to use Ctrl+c to shut down the node. We should keep that behavior as is.

### CODE REVIEW

File: cmd/harmony/main.go [hunk 1]
```diff
@@ -645,15 +645,10 @@ func main() {
 		for {
 			select {
 			case sig := <-osSignal:
-				if sig == os.Kill || sig == syscall.SIGTERM {
+				if sig == os.Kill || sig == syscall.SIGTERM || sig == os.Interrupt {
 					fmt.Printf("Got %s signal. Gracefully shutting down...\n", sig)
```
Comment (fxfactorial) [inline cmd/harmony/main.go]: Very little time available in interrupt , do the print after

Comment (rlan35) [reply cmd/harmony/main.go]: Interrupt won't kill the program either way. Better to print the message early to indicate what's happening.

Comment (fxfactorial) [reply cmd/harmony/main.go]: the trying to catch on os.Kill is wrong. You can't handle a signal handler on os.kill (signal -9), doesn't work on linux.

**Comment by rlan35** [comment]
> better add a user defined signal to handle the node shutdown gracefully. SO, during the rolling update, we can send the USER signal to node program before we stop it. That can leave enough time for node to handle the disk flush.

Good idea. Can you add an issue for it, combined with the script change.

**Comment by LeoHChen** [comment]
> > better add a user defined signal to handle the node shutdown gracefully. SO, during the rolling update, we can send the USER signal to node program before we stop it. That can leave enough time for node to handle the disk flush.
> 
> Good idea. Can you add an issue for it, combined with the script change.

https://github.com/harmony-one/harmony/issues/2271

@Daniel-VDM

**Pull request merged**